code/ui/swe6813-ui/src/pages/Root/Root.jsx

```diff
@@ -1,8 +1,46 @@
-export default function Root(){
+import {
+    useCallback,
+    useEffect
+} from "react";
+import { useNavigate } from "react-router-dom";
+import { checkToken } from "../../util/Api";
+import Button from '../../components/Button/Button';
+
+export default function Root() {
+    const navigate = useNavigate()
+
+    const signoutHandler = useCallback(() => {
+        localStorage.removeItem('token');
+        navigate('/auth');
+    }, [navigate])
+
+    useEffect(() => {
+        const token = localStorage.getItem('token');
+        if (!token) {
+            return
+        }
+        
+        checkToken(token)
+            .then((isValid) => {
+                if (isValid) {
+                    return
+                }
+
+                localStorage.removeItem('token')
+                navigate('/auth')
+            })
+            .catch(console.log)
```

Comment:
Let's add a TODO or issue to make some sort of generic success / error component we can put data in instead of the console.  Probably not something we'll get to but shows the forethought.

Reply:
sounds good! added TODOs, we can do a single redux action that'll render an error message once we have it set up!